src/programs/igotugui/igotugui.1
.\" First parameter, NAME, should be all caps
.\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
.\" other parameters are allowed: see man(7), man(1)
.TH IGOTUGUI 1 "October 03, 2009"
.\" Please adjust this date whenever revising the manpage.
.\"
.\" Some roff macros, for reference:
.\" .nh        disable hyphenation
.\" .hy        enable hyphenation
.\" .ad l      left justify
.\" .ad b      justify to both left and right margins
.\" .nf        disable filling
.\" .fi        enable filling
.\" .br        insert line break
.\" .sp <n>    insert n+1 empty lines
.\" for manpage-specific macros, see man(7)

.SH NAME
.PP
igotugui \- GUI the i-gotU GPS trackers

.SH SYNOPSIS
.PP
.B igotugui [\fIOPTIONS\fR]

.SH DESCRIPTION
.\" TeX users may be more comfortable with the \fB<whatever>\fP and
.\" \fI<whatever>\fP escape sequences to invoke bold face and italics,
.\" respectively.
.PP
\fBigotugui\fP gives access to the i-gotU USB GPS trackers via serial port
(Windows) or libusb (Linux) from an easy to use GUI. It is possible to show the
configuration of a GPS tracker, export tracks and waypoints in GPX format and
clear the internal flash memory.
.TP
\fB\-\-help\fR
help message
.TP
\fB\-\-version\fR
version information
.TP
\fB\-d\fR, \fB\-\-verbose\fR
show USB and data transfer debug messages

.SH AUTHOR
.PP
igotugui was written by Michael Hofmann <mh21@mh21.de>.

.SH "REPORTING BUGS"
.PP
Report bugs at https://bugs.launchpad.net/igotu2gpx/.

.SH COPYRIGHT
.PP
Copyright \(co 2009 Michael Hofmann <mh21@mh21.de>.
.PP
This is free software. You may redistribute copies of it under the terms of the
GNU General Public License <http://www.gnu.org/licenses/gpl.html>.
There is NO WARRANTY, to the extent permitted by law.
.SH SEE ALSO
http://mh21.de/igotu2gpx/


debian/changelog
igotu2gpx (0.3.0-0) UNRELEASED; urgency=low

  * Released igot2gpx 0.3.0.
  * Changes since version 0.2.94:
    - Allow http update urls again.
    - Fixed update check.
    - Updated translations.

 -- Michael Hofmann <mh21@piware.de>  Mon, 26 Oct 2009 20:40:47 +0100

igotu2gpx (0.2.94-0) UNRELEASED; urgency=low

  * Released the fourth release candidate of igotu2gpx 0.3.0.
  * Changes since version 0.2.93:
    - Fixed track downloading on newer GT100/GT200/GT200e devices

 -- Michael Hofmann <mh21@piware.de>  Mon, 05 Oct 2009 20:47:22 +0200

igotu2gpx (0.2.93-0) UNRELEASED; urgency=low

  * Released the third release candidate of igotu2gpx 0.3.0.
  * Changes since version 0.2.92:
    - Updates fixed by supplying the missing Host: header
    - Include lib32 and lib64 in plugin search
    - Option to group points into segments instead of tracks on GPX output added, manpage updates
    - Debian packaging fix on Hardy
    - Support to recognize a GT-200e
    - Do not crash if no connection plugins are found

 -- Michael Hofmann <mh21@piware.de>  Sun, 04 Oct 2009 15:39:15 +0200

igotu2gpx (0.2.92-0) UNRELEASED; urgency=low

  * Released the second release candidate of igotu2gpx 0.3.0.
  * Changes since version 0.2.91:
    - Support for GT-100
    - SSL support for update notifications
    - Dutch translation thanks to Peter Florijn
    - French translation thanks to Inti Raymi
    - Polish translation thanks to Jakub Kowalski

 -- Michael Hofmann <mh21@piware.de>  Sat, 26 Sep 2009 01:23:14 +0200

igotu2gpx (0.2.91-0) UNRELEASED; urgency=low

  * Released the first release candidate of igotu2gpx 0.3.0.
  * Changes since version 0.2.3:
    - Localization
    - Update notifications
    - Reworked GUI
    - GT120/GT200 purge support
    - Export with offset to UTC


 -- Michael Hofmann <mh21@piware.de>  Sat, 05 Sep 2009 00:25:25 +0200

igotu2gpx (0.2.2-1ubuntu1) jaunty; urgency=low

  * Linux icon fix.

 -- Michael Hofmann <mh21@piware.de>  Thu, 28 May 2009 10:45:55 +0200

igotu2gpx (0.2.2-1) jaunty; urgency=low

  * Point release.
    - complete GUI

 -- Michael Hofmann <mh21@piware.de>  Thu, 28 May 2009 10:13:49 +0200

igotu2gpx (0.2.2~rc1-1) jaunty; urgency=low

  * Point release candidate.
    - works on Mac OS X
    - better connection error handling
    - rudimentary GUI

 -- Michael Hofmann <mh21@piware.de>  Fri, 08 May 2009 10:11:04 +0200

igotu2gpx (0.2.1-1) jaunty; urgency=low

  * Point release.
    - builds on Mac OS X

 -- Michael Hofmann <mh21@piware.de>  Thu, 30 Apr 2009 14:04:27 +0200

igotu2gpx (0.2-1) intrepid; urgency=low

  * New upstream release.
    - complete track point decoding (with --details)
    - configuration decoding mostly complete
    - diff mode to analyze configuration changes
    - reliable Windows serial driver

 -- Michael Hofmann <mh21@piware.de>  Sat, 04 Apr 2009 22:55:08 +0200

igotu2gpx (0.1.1-1) intrepid; urgency=low

  * New upstream release. Much improved usb transport.

 -- Michael Hofmann <mh21@piware.de>  Thu, 26 Mar 2009 08:19:54 +0100

igotu2gpx (0.1-2) intrepid; urgency=low

  * Packaging fix: include libusb-dev dependency.

 -- Michael Hofmann <mh21@piware.de>  Tue, 24 Mar 2009 23:43:48 +0100

igotu2gpx (0.1-1) intrepid; urgency=low

  * Initial release.

 -- Michael Hofmann <mh21@piware.de>  Tue, 24 Mar 2009 23:19:53 +0100


src/programs/igotugui/igotugui.cpp
/******************************************************************************
 * Copyright (C) 2009  Michael Hofmann <mh21@mh21.de>                         *
 *                                                                            *
 * This program is free software; you can redistribute it and/or modify       *
 * it under the terms of the GNU General Public License as published by       *
 * the Free Software Foundation; either version 3 of the License, or          *
 * (at your option) any later version.                                        *
 *                                                                            *
 * This program is distributed in the hope that it will be useful,            *
 * but WITHOUT ANY WARRANTY; without even the implied warranty of             *
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the              *
 * GNU General Public License for more details.                               *
 *                                                                            *
 * You should have received a copy of the GNU General Public License along    *
 * with this program; if not, write to the Free Software Foundation, Inc.,    *
 * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.                *
 ******************************************************************************/

#include "igotu/commonmessages.h"
#include "igotu/messages.h"
#include "igotu/optioncontext.h"
#include "igotu/paths.h"

#include "iconstorage.h"
#include "mainwindow.h"

#include <boost/shared_ptr.hpp>

#include <QApplication>
#include <QFileInfo>
#include <QLibraryInfo>
#include <QLocale>
#include <QTranslator>

using namespace igotu;

// Put translations in the right context
//
// TRANSLATOR igotu::Common

int main(int argc, char *argv[])
{
    QApplication app(argc, argv);
    app.setApplicationName(QLatin1String("igotugui"));
    app.setOrganizationName(QLatin1String("mh21.de"));
    app.setOrganizationDomain(QLatin1String("mh21.de"));
    app.setWindowIcon(IconStorage::get(IconStorage::GuiIcon));
    // clean menus
    app.setAttribute(Qt::AA_DontShowIconsInMenus);

    QTranslator qtTranslator;
    qtTranslator.load(QLatin1String("qt_" )+ QLocale::system().name(),
            QLibraryInfo::location(QLibraryInfo::TranslationsPath));
    app.installTranslator(&qtTranslator);

    // TODO: add marble translations

    QList<boost::shared_ptr<QTranslator> > translators;
    QListIterator<QString> path(Paths::translationDirectories());
    for (path.toBack(); path.hasPrevious();) {
        boost::shared_ptr<QTranslator> translator(new QTranslator);
        if (!translator->load(QLatin1String("igotu2gpx_") +
                    QLocale::system().name(), path.previous()))
            continue;
        translators.append(translator);
        app.installTranslator(translator.get());
    }

    bool version = false;
    int verbose = 0;

    OptionContext context(app.arguments(),
            MainWindow::tr("[OPTION...]"),
            OptionGroup(QString(), Common::tr("Program Options"), QString(),
                QString(), QList<OptionEntry>()
            << OptionEntry(QLatin1String("version"), 0, 0,
                 OptionEntry::NoArgument, &version,
                 Common::tr("output version information and exit"))
            << OptionEntry(QLatin1String("verbose"), 0, 0,
                 OptionEntry::NoArgument, incrementOptionValue(&verbose),
                 Common::tr("increase verbosity"))));

    try {
        QStringList additionalParams = context.parse();

        if (version) {
            Messages::textOutput(Common::tr(
                    "<h3>Igotu2gpx %1</h3><br/>"
                    "Downloads tracks and waypoints from MobileAction i-gotU USB GPS travel loggers.<br/><br/>"
                    "Copyright (C) 2009 Michael Hofmann.<br/>"
                    "License GPLv3+: GNU GPL version 3 or later (http://gnu.org/licenses/gpl.html)<br/>"
                    "This is free software: you are free to change and redistribute it.<br/>"
                    "There is NO WARRANTY, to the extent permitted by law.")
                    .replace(QRegExp(QLatin1String("<br/>")), QLatin1String("\n"))
                    .remove(QRegExp(QLatin1String("<[^>]+>")))
                    .arg(QLatin1String(IGOTU_VERSION_STR)));
            return 0;
        }
    } catch (const std::exception &e) {
        Messages::errorMessage(Common::tr("Unable to parse command line parameters: %1")
                    .arg(QString::fromLocal8Bit(e.what())));
        return 2;
    }

    Messages::setVerbose(verbose);

    MainWindow mainWindow;
    mainWindow.show();
    return app.exec();
}
